### ISSUE TITLE
Cluster unavailable after reboot

### ISSUE THREAD

**Issue by pensiero** [rancher/rancher#19678]
After rebooting the server where the Rancher agent runs, the Rancher UI tells me that the Cluster is unavailable.

I entered inside the server where the agent runs, and discovered that the agent didn't started after reboot. So I run the command to setup a new agent, the agent container is up, however still the cluster is unavailable.

```
docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.1.4 ......... --etcd --controlplane --worker
```

Agent and Server are running on 2 different machines.

**Comment by pensiero** [comment]
Ok seems that the agent crashes, this is the error log:

```
INFO: Arguments: -- share-root.sh docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.2.2 --server https://furia.oscarfanelli.com --token REDACTED --ca-checksum 3702d8dbae9e1c1048ce9bda7b050544d8cc3a63c6235bf8dec407fa79c5d1c9 --no-register --only-write-certs /var/lib/kubelet /var/lib/rancher
+ trap 'exit 0' SIGTERM
++ grep :devices: /proc/self/cgroup
++ head -n1
++ awk -F/ '{print $NF}'
++ sed -e 's/docker-\(.*\)\.scope/\1/'
+ ID=3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
++ docker inspect -f '{{.Config.Image}}' 3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
+ IMAGE=rancher/rancher-agent:v2.2.2
+ bash -c 'docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.2.2 --server https://furia.oscarfanelli.com --token vr5bvvfb8nmz2mk2sr2ndclp7885zhj4dwrdv2nm9xvmzpkpn76rh8 --ca-checksum 3702d8dbae9e1c1048ce9bda7b050544d8cc3a63c6235bf8dec407fa79c5d1c9 --no-register --only-write-certs'
c972226d5f99abab7e186fa2aedcf48818e5561b185e7cb93b01f8198d68982b
+ docker run --privileged --net host --pid host -v /:/host --rm --entrypoint /usr/bin/share-mnt rancher/rancher-agent:v2.2.2 /var/lib/kubelet /var/lib/rancher -- norun
+ docker start kubelet
kubelet
+ docker kill --signal=SIGTERM 3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
++ exit 0
INFO: Arguments: -- share-root.sh docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.2.2 --server https://furia.oscarfanelli.com --token REDACTED --ca-checksum 3702d8dbae9e1c1048ce9bda7b050544d8cc3a63c6235bf8dec407fa79c5d1c9 --no-register --only-write-certs /var/lib/kubelet /var/lib/rancher
+ trap 'exit 0' SIGTERM
++ grep :devices: /proc/self/cgroup
++ head -n1
++ awk -F/ '{print $NF}'
++ sed -e 's/docker-\(.*\)\.scope/\1/'
+ ID=3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
++ docker inspect -f '{{.Config.Image}}' 3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
+ IMAGE=rancher/rancher-agent:v2.2.2
+ bash -c 'docker run -d --privileged --restart=unless-stopped --net=host -v /etc/kubernetes:/etc/kubernetes -v /var/run:/var/run rancher/rancher-agent:v2.2.2 --server https://furia.oscarfanelli.com --token vr5bvvfb8nmz2mk2sr2ndclp7885zhj4dwrdv2nm9xvmzpkpn76rh8 --ca-checksum 3702d8dbae9e1c1048ce9bda7b050544d8cc3a63c6235bf8dec407fa79c5d1c9 --no-register --only-write-certs'
9a0ca1d41dae889ca50840e7e71d1d4200131d080881adfd8004fc1b0c4aa13a
+ docker run --privileged --net host --pid host -v /:/host --rm --entrypoint /usr/bin/share-mnt rancher/rancher-agent:v2.2.2 /var/lib/kubelet /var/lib/rancher -- norun
+ docker start kubelet
kubelet
+ docker kill --signal=SIGTERM 3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
3ec09b7c34763b79cb22389d3a0c07ad59759611278733f51d4a135de5c96aed
++ exit 0
```

**Comment by pensiero** [comment]
Resolved by upgrading to `v2.2.2` !?!?!?!!?
Doesn't really know why, however it worked.

I upgraded to rancher server v2.2.2, and then I manually run the command to install the v2.2.2 agent on the agent machine.
